### PR TITLE
Added interactive list api [#172950387]

### PR DESCRIPTION
**Pull request by dougmartin**

### CODE REVIEW

File: app/assets/javascripts/lara-typescript.js [hunk 1]
```diff
@@ -29314,7 +29333,9 @@ exports.ManagedInteractiveAuthoring = function (props) {
                         React.createElement("legend", null, "Url Fragment"),
                         React.createElement("textarea", { id: formField("url_fragment").id, name: formField("url_fragment").name, defaultValue: url_fragment, onBlur: handleUrlFragmentBlur })),
                     libraryInteractive.authoring_guidance
-                        ? React.createElement("div", { dangerouslySetInnerHTML: { __html: libraryInteractive.authoring_guidance } })
+                        ? React.createElement("fieldset", null,
+                            React.createElement("legend", null, "Authoring Guidance"),
+                            React.createElement("div", { dangerouslySetInnerHTML: { __html: libraryInteractive.authoring_guidance } }))
```
Comment (kswenson) [inline app/assets/javascripts/lara-typescript.js:29338]: Not necessarily part of this PR since it's not a change, but elsewhere we've been using a [renderHTML()](https://github.com/concord-consortium/question-interactives/blob/master/src/shared/utilities/render-html.ts) utility function to replace use of `dangerouslySetInnerHTML`).

File: app/models/interactive_page.rb [hunk 2]
```diff
@@ -143,10 +147,10 @@ def add_embeddable(embeddable, position = nil, section = nil)
     end
   end
 
-  def add_interactive(interactive, position = nil, validate = true)
+  def add_interactive(interactive, position = nil, validate = true, section = INTERACTIVE_BOX)
```
Comment (scytacki) [inline app/models/interactive_page.rb:150]: Is this change necessary?

File: app/models/mw_interactive.rb [hunk 3]
```diff
@@ -152,4 +152,9 @@ def update_labbook_options
   def prompt
     name
   end
+
+  def thumbnail_url
+    # mirror managed_interactive has thumbnail_url which proxies to library_interactive
+    nil
```
Comment (scytacki) [inline app/models/mw_interactive.rb:158]: There is an image_url which should be used here if it is set.

File: lara-typescript/src/interactive-api-parent/iframe-saver.ts [hunk 4]
```diff
@@ -263,6 +266,10 @@ export class IFrameSaver {
       return this.getFirebaseJwt(request);
     });
 
+    this.addListener("getInteractiveList", (request: LaraInteractiveApi.IGetInteractiveListRequest) => {
```
Comment (scytacki) [inline lara-typescript/src/interactive-api-parent/iframe-saver.ts:269]: I think this iframe-saver is only used at runtime. The getInteractiveList api will be needed during authoring.

File: app/models/interactive_page.rb (hunk 2 above)
Comment (dougmartin) [reply app/models/interactive_page.rb:150]: Yes, this allows us pass `nil` to add the interactive to the assessment block here: https://github.com/concord-consortium/lara/pull/596/files#diff-85f32c83abf3ed472f1a05afc00bcd5eR83

Comment (scytacki) [reply app/models/interactive_page.rb:150]: It is probably better to call add_embeddable directly in that case.

This method is only used in tests. And it shows the interactive section when it is called.
Ideally we'd remove this method and move it to a test helper.

**Comment by scytacki** [comment]
I'm guessing this isn't a draft PR anymore right?
